### DISCUSSION TITLE
Using Mono.Cecil from a dotnet core app to load dotnet core and dotnet framework assemblies

### DISCUSSION THREAD

**Discussion by rouke-broersma** [jbevain/cecil#744] [Q&A] [answered]
Hi 👋 

At https://github.com/stryker-mutator/stryker-net we use Mono.Cecil to load embedded resources from dll's so we can attach them to our manual (using roslyn sdk) recompile. This because there are many embedded resource types such as asp.net translations that are generated using 'custom tools' and we don't want to have to replicate all these tools in stryker. We don't always seem to be able to load the dll's due to errors.

One user had the following issue: https://github.com/stryker-mutator/stryker-net/issues/1501 

The exact error is:

```
Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'System.Data.Entity, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
```

The user submitted a fix here: https://github.com/stryker-mutator/stryker-net/pull/1502
The fix is using a copy of the Mono.Cecil BaseAssemblyResolver with the conditional compilations removed and trying both the dotnet core and dotnet framework ways to resolve the assemblies. This does fix the issue for this user but I am hesitant to use this fix because I have no clue what the impact of these changes could potentially be.

Could anyone shine a light on if this is a good fix or if there is perhaps a better fix for this problem? I would rather want to avoid having to maintain a copy of a file from Mono.Cecil if I can.

**Comment by jbevain** [comment, marked as answer]
Hey,

At the end of the day, the resolving behavior provided by Cecil is a default one, and each application might have different requirements.

Everything is done in Cecil so that you can provide you own assembly resolver, so if you need a resolver that can resolve both .NET Core and .NET Framework assemblies, feel free to either roll out your own, or take modify Cecil's to work for your use case.
